src/parse/parser.rs

```diff
@@ -1061,11 +1061,39 @@ impl<'help, 'app> Parser<'help, 'app> {
             self.app.settings.set(AS::ValidArgFound);
             self.seen.push(opt.id.clone());
             if opt.is_set(ArgSettings::TakesValue) {
+                debug!(
+                    "Parser::parse_long_arg: Found an opt with value '{:?}'",
+                    &val
+                );
                 return self.parse_opt(&val, opt, matcher);
-            } else {
-                self.check_for_help_and_version_str(&arg)?;
-                return Ok(self.parse_flag(opt, matcher));
             }
+            debug!("Parser::parse_long_arg: Found a flag");
+            self.check_for_help_and_version_str(&arg)?;
```

Comment:
Suggestion: I think clap should emits too many values error in this situation: `--help=xxx`, which means `check_for_help_and_version_str` should be placed in the `None` branch.

Reply:
Oops! I'm sorry, I seem to have forgotten the `AS::InferLongArgs` part. I'll fix that too.

Reply:
Resolved.